examples/Azure_IoT_Central_ESP32_AzureIoTKit/AzureIoT.cpp

```diff
@@ -977,7 +988,16 @@ static int get_mqtt_client_config_for_dps(azure_iot_t* azure_iot, mqtt_client_co
 
   mqtt_client_config->client_id = client_id_span;
   mqtt_client_config->username = username_span;
-  mqtt_client_config->password = password_span;
+
```

Comment:
This code is exactly the same as above. Can't we extract it into a single file and reuse that file in both samples?

Reply:
No it's not. One generates the SAS for iot hub, and the other for DPS. The calls into the azure-sdk-for-c lib are different, thus requiring separate functions. HAving more abstraction here would only increase the complexity and not help much.
If I didn't get your idea, please clarify.

Reply:
@danewalton , however I think maybe the following block can be avoided by setting ->password up in the code as an else after line 971?